go/exosocket/index.go

```diff
@@ -0,0 +1,106 @@
+package exosocket
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/Originate/exocom/go/structs"
+	"github.com/Originate/exocom/go/utils"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/gorilla/websocket"
+	"github.com/pkg/errors"
+)
+
+// ExoSocket is the low level API to Exocom
+type ExoSocket struct {
+	config                      Config
+	shouldReconnectOnDisconnect bool
+	socket                      *websocket.Conn
+	messageChannel              chan structs.Message
+}
+
+// NewExoSocket returns an ExoSocket with the given config
+func NewExoSocket(config Config) *ExoSocket {
+	return &ExoSocket{config: config}
+}
+
+// Connect brings an ExoSocket instance online
+func (e *ExoSocket) Connect() error {
+	if e.config.Port == 0 {
+		e.config.Port = 80
+	}
+	exocomURL := fmt.Sprintf("ws://%s:%d/services", e.config.Host, e.config.Port)
```

Comment:
What if `Host` is empty here? Consider using a`url.URL{Scheme: "ws", Host: ...}`.

Reply:
updated. Don't think that changed much though as the Host and port are still passed in as a single string